corehq/apps/callcenter/sync_usercase.py

```diff
@@ -105,7 +105,10 @@ def case_should_be_reopened(case, user_case_should_be_closed):
         return case and case.closed and not user_case_should_be_closed
 
     if not case:
-        user_case_helper.create_usercase(fields)
+        if not user.to_be_deleted():
+            user_case_helper.create_usercase(fields)
+        else:
+            return user_case_helper
```

Comment:
Rather than doing the check here, what about not calling the task at all? The task is called in `user.save`: https://github.com/dimagi/commcare-hq/blob/77010c675870556682dd08ec8e7b535f07e8cba2/corehq/apps/users/models.py#L1677

So we could check there if the user is 'deleted' or not.

Reply:
Early check makes sense. Modified.

Reply:
Just wanted to confirm on this change once. 

So the usercase is already deleted hence skipping sync for usercase sounds okay but there is more happening in that function.

https://github.com/dimagi/commcare-hq/blob/d2aa0eda2f0bf89ad9be97079c323ee3f7c63495/corehq/apps/callcenter/sync_usercase.py#L259-L265

Are we good to skip `_iter_call_center_case_helpers` as well?

Reply:
yes, fine to skip the entire function